### ISSUE TITLE
libical-glib needs a C11 feature when compiling with clang

### ISSUE THREAD

**Issue by winterz** [libical/libical#543]
I got some update on Fedora 34 today, maybe it was clang?  I don't know.
but anyhow in master I'm seeing these errors when building libical-glib with clang12.0.1:

```/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-array.h:35:26: warning: redefinition of typedef 'ICalArray' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-attach.h:35:26: warning: redefinition of typedef 'ICalAttach' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-comp-iter.h:35:26: warning: redefinition of typedef 'ICalCompIter' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-component.h:39:26: warning: redefinition of typedef 'ICalComponent' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-datetimeperiod.h:37:26: warning: redefinition of typedef 'ICalDatetimeperiod' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-duration.h:35:26: warning: redefinition of typedef 'ICalDuration' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-geo.h:35:26: warning: redefinition of typedef 'ICalGeo' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-parameter.h:36:26: warning: redefinition of typedef 'ICalParameter' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-parser.h:35:26: warning: redefinition of typedef 'ICalParser' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-period.h:35:26: warning: redefinition of typedef 'ICalPeriod' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-property.h:38:26: warning: redefinition of typedef 'ICalProperty' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-recur-iterator.h:35:26: warning: redefinition of typedef 'ICalRecurIterator' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-recurrence.h:35:26: warning: redefinition of typedef 'ICalRecurrence' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-reqstat.h:36:26: warning: redefinition of typedef 'ICalReqstat' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-time.h:35:26: warning: redefinition of typedef 'ICalTime' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-time-span.h:36:26: warning: redefinition of typedef 'ICalTimeSpan' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-timezone.h:35:26: warning: redefinition of typedef 'ICalTimezone' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-trigger.h:35:26: warning: redefinition of typedef 'ICalTrigger' is a C11 feature [-Wtypedef-redefinition]
/home/allen/projects/libical/libical/build-testclang1-clang/src/libical-glib/i-cal-value.h:38:26: warning: redefinition of typedef 'ICalValue' is a C11 feature [-Wtypedef-redefinition]```

**Comment by winterz** [comment]
I did add this code to CMakeLists.txt a couple weeks ago.  I'll try removing.
```# C99 compliant compiler is required
set(CMAKE_C_STANDARD 99)
set(CMAKE_C_STANDARD_REQUIRED ON)
```

**Comment by winterz** [comment]
should be fixed in f109855e238f7857d4683be838e8acd5fcafd0a2
in libical-glib/CMakeLists.txt I added:
```
set(CMAKE_C_STANDARD 11)
set(CMAKE_C_STANDARD_REQUIRED ON)
```

IOW: C99 is good enough for everything except libical-glib which needs C11
